nearcore/src/cold_storage.rs

```diff
@@ -117,8 +135,66 @@ fn cold_store_copy_result_to_string(result: &anyhow::Result<ColdStoreCopyResult>
     }
 }
 
+#[derive(Debug)]
+enum ColdStoreInitialMigrationResult {
+    /// Cold storage was already initialized
+    NoNeedForMigration,
+    /// Performed a successful cold storage migration
+    SuccessfulMigration,
+    /// Migration was interrupted by keep_going flag
+    MigrationInterrupted,
+}
+
+/// This function performes intial population of cold storage if needed.
+/// Migration can be interrupted via `keep_going` flag.
+///
+/// First, checks that hot store is of kind `Archive`. If not, no migration needed.
+/// Then, captures hot final head BEFORE the migration, as migration is performed during normal neard run.
+/// If hot final head is not set, returns Err.
+/// Otherwise:
+/// 1. performed migration
+/// 2. updates head to saved hot final head
+///
+/// Any Ok status means that this function should not be retried:
+/// - either migration was performed (now or earlier)
+/// - or migration was interrupted, which means the `keep_going` flag was set to `false`
+///   which means that everything cold store thread related has to stop
```

Comment:
If the migration is interrupted what does it mean for the node owner? Can they safely restart neard and will migration restart / resume and work as expected?

Reply:
Yes, interruption is safe. Right now they may want to empty cold-data dir first, I don't really know the efficiency of setting same value to the same key multiple times in rocksdb. BTW, if it does consume too much space, we will hit it during testing with the State column and will need to optimise it in one block copying.